packages/openapi3/test/tsp-openapi3/output/polymorphism/main.tsp

```diff
@@ -0,0 +1,45 @@
+import "@typespec/http";
+import "@typespec/openapi";
+import "@typespec/openapi3";
+
+using Http;
+using OpenAPI;
+
+@service({
+  title: "Polymorphism sample",
+})
+@info({
+  version: "0.0.0",
+})
+namespace Polymorphismsample;
+
+model Cat extends Pet {
+  kind: "cat";
+  meow: int32;
+}
+
+model Dog extends Pet {
+  kind: "dog";
+  bark: string;
+}
+
+@discriminator("kind")
+model Pet {
+  name: string;
+  weight?: float32;
+
+  /**
+   * Discriminator property for Pet.
+   */
+  kind?: string;
+}
+
+/**
+ * The request has succeeded.
+ */
+model root_read200ApplicationJsonResponse {
+  @statusCode statusCode: 200;
+  @bodyRoot body: Pet;
+}
+
+@route("/Pets") @get op root_read(): root_read200ApplicationJsonResponse;
```

Comment:
I think we should have an interface called `root` and then an operation called `read` inside that interface.

Reply:
Is your reasoning here based on the presence of the underscore in the operation id? I originally thought to create an interface in such a scenario, but then thought this might be a heuristic 3rd parties would want to opt into.

The [OpenAPI3 spec](https://swagger.io/specification/) says about `operationId`:
> it is RECOMMENDED to follow common programming naming conventions

I imagined that if the primary language a spec author wrote in preferred snake_case (python/ruby?), creating this interface by default would be incorrect.

Reply:
So roundtrip correctness is not our goal? Is there any case that will generate interface/namespace?

Reply:
I think that might be a good candidate for having an option for specifying the separator, as some people use `.` too and I assume many other options.

```
tsp-openapi3 convert foo.json --group-operation-sep=_
```

Reply:
but I think all those things can be done as follow up as the goal here is to get something out quick. I don't think we should expect the tsp to be generated to be something a user would keep as it without any sort of cleanup anyway.

Reply:
> So roundtrip correctness is not our goal?
Correct - this tool is meant for migrating to TypeSpec from OpenAPI3 - where some of these conventions (e.g. prefixing operationId with the interface name) would not be used. As @timotheeguerin mentions we would expect users to cleanup the generated tsp as they see fit once we give them this starting point.